Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/HeatExchagerWithInputEffectiveness.mo

```diff
@@ -0,0 +1,185 @@
+            -100},{100,100}}), graphics={
+        Rectangle(
+          extent={{-70,80},{70,-80}},
+          lineColor={0,0,255},
+          pattern=LinePattern.None,
+          fillColor={0,62,0},
+          fillPattern=FillPattern.Solid),
+        Text(
+          extent={{-62,50},{48,-10}},
+          textColor={255,255,255},
+          textString="epsS=%epsS"),
+        Text(
+          extent={{-60,4},{50,-56}},
+          textColor={255,255,255},
+          textString="epsL=%epsL")}),
+          preferredView="info",
+defaultComponentName="hexInpEff",
+Documentation(info="<html>
+<p>
+This block is identical to
+<a href=\"modelica://Buildings.Fluid.MassExchangers.ConstantEffectiveness\">
+Buildings.Fluid.MassExchangers.ConstantEffectivenesst</a>,
+except that the effectiveness are inputs rather than parameters.
+</p>
+<p>
+This model transfers heat and moisture in the amount of
+</p>
+<pre>
+  QSen = epsS * Q_max,
+  m    = epsL * mWat_max,
+</pre>
+<p>
+where <code>epsS</code> and <code>epsL</code> are input effectiveness
+for the sensible and latent heat transfer,
+<code>Q_max</code> is the maximum sensible heat that can be transferred and
+<code>mWat_max</code> is the maximum moisture that can be transferred.
+</p>
+<p>
+This model can only be used with medium models that define the integer constant
+<code>Water</code> which needs to be equal to the index of the water mass fraction
+in the species vector.
+</p>
+</html>",
+revisions="<html>
+<ul>
+<li>
+September 29, 2023, by Sen Huang:<br/>
+Changing the effectiveness from parameters to inputs.
+</li>
+<li>
+April 30, 2018, by Filip Jorissen:<br/>
+Set <code>final prescribedHeatFlowRate1=true</code> and 
+<code>final prescribedHeatFlowRate2=true</code>.<br/>
+See
+<a href=\"https://github.com/ibpsa/modelica-ibpsa/issues/907\">#907</a>.
+</li>
+<li>
+April 11, 2017, by Michael Wetter:<br/>
+Corrected bug as <code>Q1_flow</code> did not include latent heat flow rate.<br/>
+This is for issue
+<a href=\"https://github.com/lbl-srg/modelica-buildings/issues/704\">Buildings #704</a>.
+</li>
+<li>
+October 14, 2013 by Michael Wetter:<br/>
+Replaced access to constant <code>Medium1.Water</code> by introducing
+the parameter <code>i1_w</code>, and used a similar construct for
+<code>Medium2</code>.
+This avoids an error during model check as these constants are not known
+in the partial medium model.
+</li>
+<li>
+August 13, 2013 by Michael Wetter:<br/>
+Corrected error in the documentation.
+</li>
+<li>
+July 30, 2013 by Michael Wetter:<br/>
+Updated model to use new variable <code>mWat_flow</code>
+in the base class.
+</li>
+<li>
+January 28, 2010, by Michael Wetter:<br/>
+Added regularization near zero flow.
+</li>
+<li>
+October 21, 2008, by Michael Wetter:<br/>
+First implementation, based on
+<a href=\"modelica://Buildings.Fluid.HeatExchangers.ConstantEffectiveness\">
+Buildings.Fluid.HeatExchangers.ConstantEffectiveness</a>.
+</li>
+</ul>
```

Comment:
That is a new model which did not start in 2008. PLEASE review your code before it is sent for review.